### PR TITLE
Feat(bigquery): pushdown CTE column names

### PR DESCRIPTION
**Pull request by GeorgeSittas**

### CODE REVIEW

File: sqlglot/dialects/bigquery.py [hunk 1]
```diff
@@ -104,6 +107,33 @@ def _unqualify_unnest(expression: exp.Expression) -> exp.Expression:
...
+        column_names = expression.alias_column_names
+
+        if cte_query.is_star:
+            logger.warning(
+                "Can't push down CTE column names for star queries. Run the query through"
+                " the optimizer or use 'qualify' to expand the star projections first."
+            )
+            return expression
+
+        expression.args["alias"].set("columns", None)
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:123]: expression.args["alias"].pop("columns")

Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py:123]: Will this work with `columns` being a list?

**Review by tobymao — APPROVED** [review]

File: sqlglot/dialects/bigquery.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/dialects/bigquery.py:123]: aren't we just getting rid of columns?

Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py:123]: Maybe we can update the API of `pop` so that it receives a `key` arg to set to `None`.

Comment (GeorgeSittas) [reply sqlglot/dialects/bigquery.py:123]: > aren't we just getting rid of columns?

Yep, it's just that this API is not currently supported. No big issue though, I can just add an optional arg in `pop`.

**Pull request merged**